src/server/scripts/EasternKingdoms/Scholomance/boss_kirtonos_the_herald.cpp

```diff
@@ -187,25 +191,39 @@ class boss_kirtonos_the_herald : public CreatureScript
                 case EVENT_DOMINATE_MIND:
                     if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1, 20.0f, true))
                         me->CastSpell(target, SPELL_DOMINATE_MIND, false);
-                    events.ScheduleEvent(EVENT_DOMINATE_MIND, urand(44000, 48000));
                     break;
                 case EVENT_KIRTONOS_TRANSFORM:
-                    if (me->HealthBelowPct(50))
+                    TransformsCount++;
+
+                    if (me->HasAura(SPELL_KIRTONOS_TRANSFORM))
                     {
                         events.Reset();
                         events.ScheduleEvent(EVENT_SWOOP, 4000);
                         events.ScheduleEvent(EVENT_WING_FLAP, 7000);
                         events.ScheduleEvent(EVENT_PIERCE_ARMOR, 11000);
                         events.ScheduleEvent(EVENT_DISARM, 15000);
+                        // show shape-shift animation before aura removal
+                        me->CastSpell(me, SPELL_TRANSFORM_VISUAL, true);
                         me->RemoveAura(SPELL_KIRTONOS_TRANSFORM);
                         me->SetUInt32Value(UNIT_VIRTUAL_ITEM_SLOT_ID + 0, uint32(0));
-                        break;
+                    } else {
+                        events.Reset();
+                        events.ScheduleEvent(EVENT_SHADOW_BOLT_VOLLEY, 2000);
+                        events.ScheduleEvent(EVENT_CURSE_OF_TONGUES, 6000);
+                        events.ScheduleEvent(EVENT_WING_FLAP, 13000);
+                        me->CastSpell(me, SPELL_KIRTONOS_TRANSFORM, true);
+                        me->SetUInt32Value(UNIT_VIRTUAL_ITEM_SLOT_ID + 0, uint32(WEAPON_KIRTONOS_STAFF));
+
+                        // Schedule Dominate Mind on every 2nd caster transform
+                        if ((TransformsCount - 2) % 4 == 0)
+                            events.ScheduleEvent(EVENT_DOMINATE_MIND, urand(4000, 8000));
                     }
 
-                    events.ScheduleEvent(EVENT_KIRTONOS_TRANSFORM, 2000);
+                    events.ScheduleEvent(EVENT_KIRTONOS_TRANSFORM, 20000);
                     break;
             }
 
+            DoZoneInCombat();
```

Comment:
Is DoZoneInCombat a good solution since its being called all the time?
Added it because we want to make sure that player will remain in combat after dominate mind threat drops.
Without it after dominate mind player can easily restore his lost stats (e.g. sit and drink).